File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        int n=nums.length;
        if(n<=1)return true;
        int pair_count=0;
        for(int i=1;i<n;i++){
            if(nums[i]<nums[i-1])pair_count++;
        }
        if(nums[0]<nums[n-1])pair_count++;
        return pair_count<=1;
    }
}

File: 1752-check-if-array-is-sorted-and-rotated/1752-check-if-array-is-sorted-and-rotated.cpp
class Solution {
public:
    bool check(vector<int>& nums) {
        int n = nums.size();
        if(n<=1)return true;

        int pair=0;
        for(int i=1;i<n;i++){
            if (nums[i]<nums[i-1] ) {
                pair++;
            }
        }
        if (nums[0]<nums[n-1] )pair++;
        return pair <=1;
    }
};